### PR TITLE
Hit reactions

### PR DESCRIPTION
**Pull request by davipatury**
IsShield -> hit_reaction

Isso permite o uso de random chance de bloqueio customizada para itens específicos.

### Changes
 - Grenade have a chance to explode when the person holding them is shot
 - Flamethrowers can release their tank's contents if the person holding them is shot
 - Can no longer self-trigger shield checks
 - Hugging someone/shaking them up bypasses shield checks (oh no, hugs of death)
 - Riot shields have a +30% chance to block thrown projectiles
 - Hunter pounces are now a thrown attack as far as shield go, and yes, can be blocked by riot shields now
 - Energy shields no longer block anything other than beam and energy projectiles
 - Being able to be pushed is based on the block chance of the item you're holding instead of it just being based on riot shields

### Fixes
 - Various armor penetration fixes, either not being respected or not being at the right location

### Code Stuffs
 - Throwing is now casted to items instead of objects. I can't think of a single non-item object you can get ahold of, and this allows thrown items to adhere to armor penetration
 - Removed unused flags
 - Simple Animals can now have armor penetration
 - Makes the reactive armor's getting EMP'd actually update the wearer's mob worn sprite

### Additions
 - Adds in HoS's hardsuit
 - Implements shielded hardsuits
  - These are basically hardsuits that can absorb hits, negating the damage and such entirely--non-accessible, for now

### CODE REVIEW

**Comment by nopm** [comment]
@davipatury PRs devem ser atômicos.

**Comment by astralenigma** [comment]
Aposto que depois deste update vamos ver ainda mais pessoas com fato espacial dentro da estação 😒

File: code/game/atoms.dm [hunk 1]
```diff
@@ -364,9 +364,6 @@
 //returns 1 if made bloody, returns 0 otherwise
 /atom/proc/add_blood(mob/living/carbon/human/M as mob)
 
-	if(flags & NOBLOODY)
-		return 0
-
```
Comment (astralenigma) [inline code/game/atoms.dm]: De certeza que este If não é importante?

**Comment by davipatury** [comment]
@nopm Todas as mudanças no PR tem relação entre si.

File: code/game/atoms.dm (hunk 1 above)
Comment (astralenigma) [inline code/game/atoms.dm]: Se bem percebo essa flag previne de ficares ensanguentado ou algo parecido.

Comment (davipatury) [inline code/game/atoms.dm]: Era usado nos telecomms mas não tinha utilidade nenhuma.

Comment (davipatury) [inline code/game/atoms.dm]: Só removi as flags não utilizadas, todas as outras continuam em seu lugar.

**Pull request merged**